src/v/cloud_topics/extent_meta.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "cloud_topics/types.h"
+#include "model/fundamental.h"
+
+#include <fmt/core.h>
+
+namespace experimental::cloud_topics {
+
+// This struct contains extent information (location of the byte slice in the
+// object + the object uuid) and the kafka metadata (base offset and committed
+// offset).
+// Timestamps are not here because the timequery is handled in the control plane
+// and the results of this are converted to kafka offsets.
+struct extent_meta {
+    // Extent information
+    object_id id;
+    // TODO: the extent meta struct has to be updated
+    // to match the RFC.
+    first_byte_offset_t first_byte_offset;
+    byte_range_size_t byte_range_size;
+
+    // Kafka metadata
+    kafka::offset base_offset;
+    kafka::offset committed_offset;
```

Comment:
What do you mean by `committed_offset`?

Reply:
I see now this is computed later to be `base_offset + num_records - 1`. Can we call this `last_offset` so it corresponds with `last_offset_delta` in the record_batch_header? Committed makes me think of the `committed_offset` we have in raft.